pkg/operator/ceph/cluster/watcher.go

```diff
@@ -91,20 +91,18 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		// If so we don't need to reconcile, this is done to avoid double reconcile on operator restart
 		// Assume the admin key since we are watching for node status to create OSDs
 		clusterInfo := cephclient.AdminClusterInfo(cluster.Namespace)
-		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, nodeName)
+		crushName := cephclient.NormalizeCrushName(nodeName)
+		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, crushName)
```

Comment:
Why do we normalize here and not have `GetOSDOnHost` call `NormalizeCrushName`? If the `nodeName` given to `GetOSDOnHost` is a k8s node name and not a CRUSH name, then it will be turned into a CRUSH name. If it is already a CRUSH name, `NormalizeCrushName` is a noop and nothing changes, right? This seems the proper place to call `NormalizeCrushName` to me.

Does that logic make sense to you @travisn  and @leseb ?

Reply:
Yes @BlaineEXE agree with you, `GetOSDOnHost` function should be the right place/level to call the `NormalizeCrushName`.